src/js/components/campaignpage/AdvertsCatalogue.jsx

```diff
@@ -264,8 +246,13 @@ const AdPreviewCarousel = ({ads, selectedIndex, setSelected}) => {
 	</div>;
 }
 
+/**
+ * @param {Object} p
+ * @param {boolean} p.active Is this the widget that the user is looking at? false if its just one in a list.
+ */
 const AdvertCard = ({ ad, active }) => {
     const size = 'landscape';
+	// ??doc - why do we have hasShown?? Is this for efficiency/performance??
     const [hasShown, setHasShown] = useState(false);
```

Comment:
Please add a doc note for our future selves - thanks. Why do we have hasShown? Is this for efficiency/performance?